terrain/Terrain.gd

```diff
@@ -61,8 +61,20 @@ func find_path_by_cell(start_cell, end_cell):
 func update_weight(unit):
 	for y in range(HEIGHT):
 		for x in range(WIDTH):
+			var currentCell = Vector2(x, y)
 			var id = flatten(x, y)
 			var cost =  unit.get_movement_cost(tiles[id].terrain_type)
+			var otherUnit = get_node("../..").get_unit_at_cell(currentCell)
```

Comment:
Maybe it would be better if we gave every unit a reference to the game instead?
Especially with units, they might not always be exaclty the "grandchild" of the game, thus get_node("../..") might not always apply. This is actually something I should have an eye on in other places as well. This locks us down on flexibility in the node hierarchy.

Reply:
Well, or in that case, just have a variable current_tile in units that updates when the unit moves and always contains the current tile they are standing on.